ast-checks/src/comment_finders.rs

```diff
@@ -0,0 +1,43 @@
+use crate::tree_utils::{next_siblings, prev_siblings, walk_kind};
+use rnix::{NodeOrToken, SyntaxElement, SyntaxKind::*, SyntaxNode};
+
+pub fn find_comment_within(node: &SyntaxNode) -> Option<String> {
+    // find the first TOKEN_COMMENT within
+    let tok = walk_kind(node, TOKEN_COMMENT).find_map(|item| match item {
+        NodeOrToken::Token(ref _tok) => Some(item),
+        NodeOrToken::Node(_) => {
+            unreachable!(
+                "`tok` should always be of kind==TOKEN_COMMENT,
+                which means that it's a SyntaxToken rather than
+                SyntaxNode"
+            );
+        }
+    })?;
```

Comment:
Maybe this can just be a `walk_kind(...).filter_map(|item| item.into_token()).take(1)`?

Reply:
Done.